### ISSUE TITLE
warning in package.json during build

### ISSUE THREAD

**Issue by ST-DDT** [faker-js/faker#3078]
There is a warning in the package.json when the code is build using tsup.

- https://github.com/faker-js/faker/actions/runs/10638166909/job/29493558767

````txt
Warning: G] The condition "default" here will never be used as it comes after both "import" and "require" [package.json]

    package.json:48:6:
      48 │       "default": "./dist/index.js"
         ╵       ~~~~~~~~~

  The "import" condition comes earlier and will be used for all "import" statements:

    package.json:46:6:
      46 │       "import": "./dist/index.js",
         ╵       ~~~~~~~~

  The "require" condition comes earlier and will be used for all "require" calls:

    package.json:47:6:
      47 │       "require": "./dist/index.cjs",
         ╵       ~~~~~~~~~

Warning: G] The condition "default" here will never be used as it comes after both "import" and "require" [package.json]

    package.json:54:6:
      54 │       "default": "./dist/locale/*.js"
         ╵       ~~~~~~~~~

  The "import" condition comes earlier and will be used for all "import" statements:

    package.json:52:6:
      52 │       "import": "./dist/locale/*.js",
         ╵       ~~~~~~~~

  The "require" condition comes earlier and will be used for all "require" calls:

    package.json:53:6:
      53 │       "require": "./dist/locale/*.cjs",
         ╵       ~~~~~~~~~
````

Related code:

https://github.com/faker-js/faker/blob/17f570e05ea413af691bb81b61dd8c28130a47e1/package.json#L44-L49
https://github.com/faker-js/faker/blob/17f570e05ea413af691bb81b61dd8c28130a47e1/package.json#L50-L55

**Comment by Shinigami92** [comment]
We should be a bit careful about this.
https://nodejs.org/api/packages.html#conditional-exports is showing that there are other environments like `"node"`. I also heard once about `"browser"` to be a thing.

So if this warning is really by `tsup`, it **might** not be fully true 🤔 

**Edit:** Or in other words: please double check if we can safely rewrite this

**Edit 2 / Afterthoughts:** maybe there is an option to suppress this warning? Or it will be gone in v10 when we only support `esm` only 🤔

**Comment by ST-DDT** [comment]
Or is this sometging that we have to report to/ask tsup?

**Comment by Shinigami92** [comment]
There is already an open issue in esbuild https://github.com/evanw/esbuild/issues/3887
- https://github.com/evanw/esbuild/issues/3887

**Comment by ST-DDT** [comment]
Fixed in #3093
